api/v1/atlasipaccesslist_types.go

```diff
@@ -0,0 +1,110 @@
+
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/api"
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/api/v1/status"
+)
+
+func init() {
+	SchemeBuilder.Register(&AtlasIPAccessList{}, &AtlasIPAccessListList{})
+}
+
+// +kubebuilder:validation:XValidation:rule="(has(self.externalProjectRef) && !has(self.projectRef)) || (!has(self.externalProjectRef) && has(self.projectRef))",message="must define only one project reference through externalProjectRef or projectRef"
+// +kubebuilder:validation:XValidation:rule="(has(self.externalProjectRef) && has(self.connectionSecret)) || !has(self.externalProjectRef)",message="must define a local connection secret when referencing an external project"
+
+// AtlasIPAccessListSpec defines the desired state of AtlasIPAccessList.
+type AtlasIPAccessListSpec struct {
+	// ProjectReference is the dual external or kubernetes reference with access credentials
+	ProjectDualReference `json:",inline"`
+	// Entries is the list of IP Access to be managed
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinItems=1
+	Entries []IPAccessEntry `json:"entries"`
+}
+
+// +kubebuilder:validation:XValidation:rule="!(has(self.ipAddress) && (has(self.cidrBlock) || has(self.awsSecurityGroup))) && !(has(self.cidrBlock) && has(self.awsSecurityGroup))",message="Only one of ipAddress, cidrBlock, or awsSecurityGroup may be set."
+
+type IPAccessEntry struct {
+	// Entry using an IP address in this access list entry.
+	// +optional
+	IPAddress string `json:"ipAddress,omitempty"`
+	// Range of IP addresses in CIDR notation in this access list entry.
+	// +optional
+	CIDRBlock string `json:"cidrBlock,omitempty"`
+	// Unique identifier of AWS security group in this access list entry.
+	// +optional
+	AwsSecurityGroup string `json:"awsSecurityGroup,omitempty"`
+	// Date and time after which Atlas deletes the temporary access list entry.
+	// +kubebuilder:validation:Type=string
+	// +kubebuilder:validation:Format=date-time
+	// +optional
+	DeleteAfterDate *metav1.Time `json:"deleteAfterDate,omitempty"`
+	// Comment associated with this access list entry.
+	// +optional
+	Comment string `json:"comment,omitempty"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+// +groupName:=atlas.mongodb.com
+// +kubebuilder:resource:categories=atlas,shortName=aial
+// +kubebuilder:printcolumn:name="Ready",type=string,JSONPath=`.status.conditions[?(@.type=="Ready")].status`
+
+// AtlasIPAccessList is the Schema for the atlasipaccesslists API.
+type AtlasIPAccessList struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   AtlasIPAccessListSpec          `json:"spec,omitempty"`
+	Status status.AtlasIPAccessListStatus `json:"status,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+
+// AtlasIPAccessListList contains a list of AtlasIPAccessList.
+type AtlasIPAccessListList struct {
```

Comment:
blocking: i believe this is not necessary. We have
1. `AtlasIPAccessList` as the top-level CRD
2. `AtlasIPAccessListSpec` as its spec
3. `IPAccessEntry` as its entries

Reply:
same as https://github.com/mongodb/mongodb-atlas-kubernetes/pull/2027#discussion_r1919961704, moving on ;-)